### PR TITLE
imag: init at 0.10.1

### PR DESCRIPTION
**Pull request by minijackson**
###### Motivation for this change

Based on #77362 by @filalex77 

As discussed there, the differences are:

- Update to 0.10.1, which removes the `Cargo.lock` patch
- Run the tests under a created PTY to make them pass
- Install the shell completion scripts

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/imag/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    # a PTY for us.
+    script -qfec "cargo test --workspace"
+  '';
+
+  preFixup = ''
+    mkdir -p "$out/share/"{bash-completion/completions,fish/vendor_completions.d,zsh/site-functions}
+    cp target/imag.bash "$out/share/bash-completion/completions/"
+    cp target/imag.fish "$out/share/fish/vendor_completions.d/"
+    cp target/_imag     "$out/share/zsh/site-functions/"
+  '';
```
Comment (filalex77) [inline pkgs/applications/misc/imag/default.nix:51]:
```suggestion
  postInstall = ''
    installShellCompletion target/imag.{bash,fish} --zsh target/_imag
  '';
```

Also add `installShellFiles` to derivation arguments and `nativeBuildInputs`.

**Review by Br1ght0ne — CHANGES_REQUESTED** [review]

**Comment by filalex77** [comment]
Checking Darwin build.

@GrahamcOfBorg build imag

**Comment by minijackson** [comment]
Updated with the usage of installShellFiles (I didn't know about this hook, thanks!), and hopefully fixes the darwin build, which uses BSD-style `script` command.

**Review by Br1ght0ne — APPROVED** [review]
diff LGTM
`nixpkgs-review` passed on NixOS 20.03

**Comment by filalex77** [comment]
@GrahamcOfBorg build imag

**Comment by minijackson** [comment]
The darwin build now passes, but I don't see the tests in the log...

**Comment by filalex77** [comment]
@GrahamcOfBorg build imag.passthru.tests

**Comment by minijackson** [comment]
@filalex77 Isn't `passthru.tests` usually for nixos modules tests?

I can see the tests in the linux build log, so I thought it was the BSD-style `script` that captured the output? Another point that bugs me is that I don't know if the `script` in the darwin build returns the same error code as the command it runs, so it's possible we might not see the darwin tests fail :-/

**Pull request merged**